subxt/src/blocks/block_types.rs

```diff
@@ -82,6 +78,12 @@ where
         ))
     }
 
+    /// Return the extrinsics associated with the block.
+    pub async fn extrinsics(&self) -> Result<extrinsics::Extrinsics<T>, Error> {
+        let block_details = self.block_details().await?;
+        extrinsics::Extrinsics::new(self.client.metadata(), block_details.block)
+    }
+
```

Comment:
I was expecting all of this extrinsics stuff to end up coming from `BlockBody`, below, with the idea that:

- you get hold of a block, and then load the body with `block.bocy().await?` 
- and then from the body you can iterate thorugh extrinsics already with `body.extrinsics()`
- and then each extrinsic you get back can be decoded (and that `Extrinsic` thing can have an interface that is inspired by `EventDetails`.

I don't think it really makes sense to have a separate top level `extrinsics` folder, because we already have a `tx` folder for constructing transactions (extrinsics) anyway, and then this `blocks` folder is all about getting and working with blocks, whose bodies are simply a list of extrinsics :)

So yeah I guess I'd maybe move the `Extrinsic` type defined already below into a separate file in this folder if it'll be getting quite big and then adding to that type the logic for decoding them (and then no need for eg an `ExtrinsicsClient` :)

Reply:
Have moved the extrinsic to the `BlocksBody` to iterate over them as `extrinsics()`

One other thing is that we still have the `find` apis that we want to mirror from the events.
Have added them with `_extrinsic` prefix as in `find_extrisic`, for extra clarity.

That makes me think that we could add something of the following:
```
struct BlockBody {
    fn extrinsics() -> Extrinsics
}

struct Extrinsics {
   fn find() ...
   fn find_first() ...
   fn iter() -> ExtrinsicDetails 
}
```

However, the `Extrinsics` struct from here would contain almost the same information as the `BlockBody` in this example. 
Let me know what you think of it 🙏